portfolio/src/main/java/com/google/sps/servlets/NewTaskServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Servlet responsible for creating new tasks. */
+@WebServlet("/new-task")
+public class NewTaskServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Sanitize user input to remove HTML tags and JavaScript.
+    String title = Jsoup.clean(request.getParameter("title"), Whitelist.none());
+    long timestamp = System.currentTimeMillis();
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Task");
+    FullEntity taskEntity =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("title", title)
+            .set("timestamp", timestamp)
+            .build();
+    datastore.put(taskEntity);
+
+    response.sendRedirect("/index.html");
```

Comment:
No need to redirect the page. This gives the feel that you're refreshing the page every time you add a new task, which can be quite bothersome.

As soon as your JavaScript code has detected that the POST request was successful, then you should feel free to display the new entity.

Reply:
Thank you, Marcus. I have learned the refreshing whole page for each change would be inefficient.
After deleting line 35, when creating a new task, a user ends up going to a blank page.
I will fix it to let users stay on the index page after creating a new task and append the created task to make it show on the page.